closed/autoconf/custom-hook.m4

```diff
@@ -19,8 +19,13 @@
 
 AC_DEFUN_ONCE([CUSTOM_EARLY_HOOK],
 [
-  JAVA_BASE_LDFLAGS="${JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
-  OPENJDK_BUILD_JAVA_BASE_LDFLAGS="${OPENJDK_BUILD_JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
+  if test "x$OPENJDK_BUILD_OS_ENV" = xwindows.cygwin; then
+    JAVA_BASE_LDFLAGS="${JAVA_BASE_LDFLAGS} -libpath:\$(SUPPORT_OUTPUTDIR)/../vm/lib"
+    OPENJDK_BUILD_JAVA_BASE_LDFLAGS="${OPENJDK_BUILD_JAVA_BASE_LDFLAGS} -libpath:\$(SUPPORT_OUTPUTDIR)/../vm/lib"
+  else
+    JAVA_BASE_LDFLAGS="${JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
+    OPENJDK_BUILD_JAVA_BASE_LDFLAGS="${OPENJDK_BUILD_JAVA_BASE_LDFLAGS} -L\$(SUPPORT_OUTPUTDIR)/../vm"
+  fi
```

Comment:
It might be clearer if we assigned `-L` or `-libpath` to a variable to be used in the flags definitions.

Reply:
I tried to follow the openjdk style - see flags.m4 
e.g.
```
if test "x$TOOLCHAIN_TYPE" = xmicrosoft; then
    $2JAVA_BASE_LDFLAGS="${$2JAVA_BASE_LDFLAGS} \
        -libpath:${OUTPUT_ROOT}/support/modules_libs/java.base"
    $2JDKLIB_LIBS=""
  else
    $2JAVA_BASE_LDFLAGS="${$2JAVA_BASE_LDFLAGS} \
        -L\$(SUPPORT_OUTPUTDIR)/modules_libs/java.base"
```

Reply:
Fair enough.

Reply:
The TOOLCHAIN_TYPE variable is not available for the EARLY_CUSTOM_HOOK (see configure.ac L102). It is set in TOOLCHAIN_DETERMINE_TOOLCHAIN_TYPE (see configure.ac L170), thus the different condition.

Reply:
Why do the Windows paths have the extra `/lib` segment?
Where is `OPENJDK_BUILD_JAVA_BASE_LDFLAGS` used?

Reply:
To set the LDFLAGS_JDKLIB

Reply:
In what file(s)?